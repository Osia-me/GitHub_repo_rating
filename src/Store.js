import React, { useState, useEffect } from "react";
import axios from 'axios';

export const Context = React.createContext(null);

export const Store = ({children}) => {
  const [state, setState] = useState([]);
  const linkToLastWeek = "https://api.github.com/search/repositories?q=created:%3E2021-06-13&sort=stars&order=desc";
  const linkToReposByLang = (lang) => `https://api.github.com/legacy/repos/search/language=${lang}`

  useEffect(() => {
    axios
      .get(linkToLastWeek)
      .then((response) => {
        setState({
          items: response.data.items,
          fav: [],
          view: null,
          language: 'Select language'
        });
      })
      .catch((error) => {
        alert('Something went wrong!')
      });
  }, []);

  useEffect(() => {
    if(state.language && state.language !== 'Select language'){
      axios.get(linkToReposByLang(state.language)).then((response) => {
        setState({
          items: response.data.repositories.map( el => ({
            name: el.name,
            description:el.description,
            html_url: el.url,
            stargazers_count: Math.round(el.score),
            id: Math.random() * 2.5
          })),
          fav: [],
          view: null,
          language: state.language
        });
      })
      .catch((error) => {
        alert('Something went wrong!')
      });
    }
  },[state.language])

  return (
    <Context.Provider value={[state, setState]}>
      { children }
    </Context.Provider>
  );
};

export default Store;
