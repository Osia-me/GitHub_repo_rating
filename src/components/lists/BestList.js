import React, { useContext } from 'react';
import Card from '../card/Card';
import { Context } from '../../Store.js';
import './BestList.css';

function BestList() {
  const [state, setState] = useContext(Context);
  const arrayOfItems = state.items;
  const favList = state.fav;

  if(!state.view){
    return (
      <div className="list">
        {arrayOfItems ? arrayOfItems.map((repo) =>
          <Card
            key={repo.id}
            id={repo.id}
            name={repo.name}
            description={repo.description}
            numberOfStars={repo.stargazers_count}
            link={repo.html_url} />
          ) : null}
      </div>
    );
  } else {
    return (
      <div className="list">
        {favList ? favList.map((repo) =>
          <Card
            key={`${repo.id}`}
            id={repo.id}
            name={repo.name}
            description={repo.description}
            numberOfStars={repo.stargazers_count}
            link={repo.html_url} />
          ) : null}
      </div>
    );
  }
}

export default BestList;
