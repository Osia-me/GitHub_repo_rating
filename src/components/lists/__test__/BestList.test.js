import Store from '../../../Store.js';
import { shallow } from 'enzyme';
import BestList from "../BestList";

test('renders BestList without crashing', () => {
  const wrapper = shallow(
    <Store className="app">
      <BestList/>
    </Store>);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains(<BestList />)).toBe(true);
});
