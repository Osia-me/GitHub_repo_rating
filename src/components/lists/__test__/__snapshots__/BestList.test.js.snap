// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BestList without crashing 1`] = `ShallowWrapper {}`;
